Tasks/ServiceFabricDeploy/deploy.ps1

```diff
@@ -0,0 +1,84 @@
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    # Load utility functions
+    . "$PSScriptRoot\utilities.ps1"
+    
+    # Collect input values
+    $publishProfilePathSearchPattern = Get-VstsInput -Name publishProfilePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForPublishProfile -ArgumentList $publishProfilePathSearchPattern) 
+    $publishProfilePath = Find-VstsFiles -LegacyPattern $publishProfilePathSearchPattern
+    Assert-SingleItem $publishProfilePath $publishProfilePathSearchPattern
+    Assert-VstsPath -LiteralPath $publishProfilePath -PathType Leaf
+    Write-Host (Get-VstsLocString -Key FoundPublishProfile -ArgumentList $publishProfilePath)
+    
+    $applicationPackagePathSearchPattern = Get-VstsInput -Name applicationPackagePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForApplicationPackage -ArgumentList $applicationPackagePathSearchPattern)
+
+    # Workaround for bug in Find-VstsFiles that doesn't handle the IncludeDirectories parameter correctly. Modifies the search pattern to look for app manifest file instead of the directory
+    # and then get the directory of that file.
+    $applicationPackagePathSearchPattern = Join-Path $applicationPackagePathSearchPattern "ApplicationManifest.xml"
+    $applicationPackagePath = Find-VstsFiles -LegacyPattern $applicationPackagePathSearchPattern
+    Assert-SingleItem $applicationPackagePath $applicationPackagePathSearchPattern
+    $applicationPackagePath = [System.IO.Path]::GetDirectoryName($applicationPackagePath)
+    Assert-VstsPath -LiteralPath $applicationPackagePath -PathType Container
+    Write-Host (Get-VstsLocString -Key FoundApplicationPackage -ArgumentList $applicationPackagePath)
+
+    $aadServiceConnectionName = Get-VstsInput -Name aadServiceConnectionName -Require
+    $aadConnectedServiceEndpoint = Get-VstsEndpoint -Name $aadServiceConnectionName -Require
+
+    $publishProfile = Read-PublishProfile $publishProfilePath
+    $clusterConnectionParameters = $publishProfile.ClusterConnectionParameters
+    
+    if (-not $clusterConnectionParameters.ContainsKey("AzureActiveDirectory") -or $clusterConnectionParameters.Item("AzureActiveDirectory") -eq $false)
+    {
+        throw (Get-VstsLocString -Key AzureActiveDirectoryNotSet)
+    }
+    
+    # Connect to cluster
+    $securityToken = Get-AadSecurityToken -ClusterConnectionParameters $clusterConnectionParameters -ConnectedServiceEndpoint $aadConnectedServiceEndpoint
+    $clusterConnectionParameters["SecurityToken"] = $securityToken
+    $clusterConnectionParameters["WarningAction"] = "SilentlyContinue"
+    [void](Connect-ServiceFabricCluster @clusterConnectionParameters)
+    Write-Host (Get-VstsLocString -Key ConnectedToCluster)
+    
+    $regKey = "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+    $moduleFolderPath = (Get-ItemProperty -Path $regKey -Name FabricSDKPSModulePath).FabricSDKPSModulePath
+
+    if (Test-Path "$moduleFolderPath\ServiceFabricSDK.ps1")
```

Comment:
This module is owned by the Service Fabric platform team and is installed on the build agents through their SDK.  It provides the actual deployment logic that we depend on.  We coordinate with them for any changes in this module so we're always aware and can react accordingly.